executor.go

```diff
@@ -796,6 +796,10 @@ func completeAbstractValue(eCtx *executionContext, returnType Abstract, fieldAST
 		runtimeType = unionReturnType.ResolveType(resolveTypeParams)
 	} else if interfaceReturnType, ok := returnType.(*Interface); ok && interfaceReturnType.ResolveType != nil {
 		runtimeType = interfaceReturnType.ResolveType(resolveTypeParams)
+		// verify the object matches the interface
+		if err := assertObjectImplementsInterface(&eCtx.Schema, runtimeType, interfaceReturnType); err != nil {
+			panic(err)
```

Comment:
Pretty wild that these are all panics, but definitely not worth refactoring now with dagql on the way.